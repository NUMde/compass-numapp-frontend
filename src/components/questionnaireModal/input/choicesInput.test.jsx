import React from 'react';
import {
  renderWithRedux,
  act,
  fireEvent,
} from '../../../../__test-utils__/render';
import ChoicesInput from './choicesInput';

const initialState = {
  Questionnaire: {
    itemMap: {
      1: {
        linkId: 1,
        type: 'group',
        item: [{ linkId: '1.1', type: 'choice', repeats: true }],
      },
      1.1: { linkId: '1.1', type: 'choice', repeats: true, answer: null },
    },
  },
};

const choiceItem = {
  linkId: '1.1',
  type: 'choice',
  repeats: true,
  required: true,
  answerOption: [
    {
      valueString: 'Option A',
    },
    {
      valueString: 'Option B',
    },
    {
      valueString: 'Option C',
    },
    {
      valueString: 'Option D',
    },
    {
      valueCoding: {
        code: 'someCode',
        system: 'someSystem',
        display: 'Option E',
      },
    },
  ],
};

describe('choicesInput (CheckBoxes)', () => {
  it('should render a list of choices as checkboxes', async () => {
    const { toJSON, getAllByTestId, getAllByText } = renderWithRedux(
      <ChoicesInput item={choiceItem} />,
      { initialState },
    );
    expect(toJSON()).toMatchSnapshot();
    expect(getAllByTestId('checkbox').length).toBe(5);
    expect(getAllByText('\uf096').length).toBe(5);
  });

  it('should render a list of integer choices as checkBoxes', async () => {
    const { getAllByTestId, getAllByText } = renderWithRedux(
      <ChoicesInput
        item={{
          ...choiceItem,
          repeats: false,
          answerOption: [
            { valueInteger: 0 },
            { valueInteger: 1 },
            { valueInteger: 2 },
          ],
        }}
      />,
      { initialState },
    );
    expect(getAllByTestId('checkbox').length).toBe(3);
    expect(getAllByText('\uf10c').length).toBe(3);
  });

  it('should check element when clicked', async () => {
    const { getByText } = renderWithRedux(<ChoicesInput item={choiceItem} />, {
      initialState,
    });
    const answerItem = getByText('Option C');
    act(() => {
      fireEvent.press(answerItem);
    });
    const selectedItem = getByText('\uf046');
    expect(selectedItem).toBeTruthy();
    expect(selectedItem.parent.parent.props.checked).toBe(true);
    expect(selectedItem.parent.parent.props.title).toBe('Option C');
    act(() => {
      fireEvent.press(selectedItem);
    });
    expect(selectedItem.props.children).not.toContain('\uf046');
  });
});

describe('choicesInput (RadioButtons)', () => {
  it('should render a list of choices as radio buttons', async () => {
    const { toJSON, getAllByTestId, getAllByText } = renderWithRedux(
      <ChoicesInput item={{ ...choiceItem, repeats: false }} />,
      { initialState },
    );
    expect(toJSON()).toMatchSnapshot();
    expect(getAllByTestId('checkbox').length).toBe(5);
    expect(getAllByText('\uf10c').length).toBe(5);
  });

  it('should render a list of integer choices as radio buttons', async () => {
    const { getAllByTestId, getAllByText } = renderWithRedux(
      <ChoicesInput
        item={{
          ...choiceItem,
          repeats: false,
          answerOption: [
            { valueInteger: 0 },
            { valueInteger: 1 },
            { valueInteger: 2 },
          ],
        }}
      />,
      { initialState },
    );
    expect(getAllByTestId('checkbox').length).toBe(3);
    expect(getAllByText('\uf10c').length).toBe(3);
  });

  it('should select answer', () => {
    const { getByText, getAllByText } = renderWithRedux(
      <ChoicesInput item={{ ...choiceItem, repeats: false }} />,
      { initialState },
    );
    const answerItem = getByText('Option C');
    act(() => {
      fireEvent.press(answerItem);
    });
    const selectedItem = getByText('\uf192');
    expect(selectedItem).toBeTruthy();
    expect(selectedItem.parent.parent.props.title).toBe('Option C');
    act(() => {
      fireEvent.press(getAllByText('\uf10c')[3]);
    });
    expect(getByText('\uf192').parent.parent.props.checked).toBe(true);
    expect(getByText('\uf192').parent.parent.props.title).toBe('Option E');
  });
});

describe('choicesInput (DropDown)', () => {
  it('should render a list of choices as dropdown', () => {
    const { toJSON, getByTestId } = renderWithRedux(
      <ChoicesInput
        item={{
          ...choiceItem,
          extension: [
            {
              url: 'http://hl7.org/fhir/StructureDefinition/questionnaire-itemControl',
              valueCodeableConcept: {
                coding: [
                  {
                    system: 'http://hl7.org/fhir/questionnaire-item-control',
                    code: 'drop-down',
                  },
                ],
              },
            },
          ],
        }}
      />,
      { initialState },
    );
    expect(toJSON()).toMatchSnapshot();
    const Picker = getByTestId('Picker');
    expect(Picker).toBeTruthy();
    expect(Picker.props.items.length).toBe(5);
  });

  it('should select answer', () => {
    const { getByTestId } = renderWithRedux(
      <ChoicesInput
        item={{
          ...choiceItem,
          extension: [
            {
              url: 'http://hl7.org/fhir/StructureDefinition/questionnaire-itemControl',
              valueCodeableConcept: {
                coding: [
                  {
                    system: 'http://hl7.org/fhir/questionnaire-item-control',
                    code: 'drop-down',
                  },
                ],
              },
            },
          ],
        }}
      />,
      { initialState },
    );

    const Picker = getByTestId('Picker');
    act(() => Picker.props.onChange({ nativeEvent: { newValue: 'Option C' } }));
    expect(Picker.props.selectedIndex).toBe(2);
  });
});
