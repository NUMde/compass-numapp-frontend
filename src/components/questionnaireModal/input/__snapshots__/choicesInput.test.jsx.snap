// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`choicesInput (CheckBoxes) should render a list of choices as checkboxes 1`] = `
Array [
  <Text
    accessibilityHint="Single Choice - only one answer option can be selected."
    style={
      Object {
        "color": "#1F2532",
        "fontFamily": "IBMPlexSans-SemiBold",
        "fontSize": 32.8125,
        "lineHeight": 42.1875,
        "marginBottom": 5,
        "marginLeft": 0,
        "marginTop": 5,
      }
    }
  />,
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 3,
        "borderWidth": 1,
        "margin": 0,
        "marginLeft": 0,
        "marginRight": 10,
        "opacity": 1,
        "padding": 10,
        "paddingLeft": 0,
        "paddingRight": 0,
      }
    }
    testID="checkbox"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#919191",
              "fontSize": 24,
            },
            Object {
              "minWidth": 24,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#1F2532",
            "fontFamily": "IBMPlexSans-SemiBold",
            "fontSize": 26.25,
            "fontWeight": "bold",
            "lineHeight": 35.625,
            "margin": 0,
            "marginLeft": 10,
            "marginRight": 10,
            "padding": 0,
          }
        }
        testID="checkboxTitle"
      >
        Option A
      </Text>
    </View>
  </View>,
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 3,
        "borderWidth": 1,
        "margin": 0,
        "marginLeft": 0,
        "marginRight": 10,
        "opacity": 1,
        "padding": 10,
        "paddingLeft": 0,
        "paddingRight": 0,
      }
    }
    testID="checkbox"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#919191",
              "fontSize": 24,
            },
            Object {
              "minWidth": 24,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#1F2532",
            "fontFamily": "IBMPlexSans-SemiBold",
            "fontSize": 26.25,
            "fontWeight": "bold",
            "lineHeight": 35.625,
            "margin": 0,
            "marginLeft": 10,
            "marginRight": 10,
            "padding": 0,
          }
        }
        testID="checkboxTitle"
      >
        Option B
      </Text>
    </View>
  </View>,
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 3,
        "borderWidth": 1,
        "margin": 0,
        "marginLeft": 0,
        "marginRight": 10,
        "opacity": 1,
        "padding": 10,
        "paddingLeft": 0,
        "paddingRight": 0,
      }
    }
    testID="checkbox"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#919191",
              "fontSize": 24,
            },
            Object {
              "minWidth": 24,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#1F2532",
            "fontFamily": "IBMPlexSans-SemiBold",
            "fontSize": 26.25,
            "fontWeight": "bold",
            "lineHeight": 35.625,
            "margin": 0,
            "marginLeft": 10,
            "marginRight": 10,
            "padding": 0,
          }
        }
        testID="checkboxTitle"
      >
        Option C
      </Text>
    </View>
  </View>,
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 3,
        "borderWidth": 1,
        "margin": 0,
        "marginLeft": 0,
        "marginRight": 10,
        "opacity": 1,
        "padding": 10,
        "paddingLeft": 0,
        "paddingRight": 0,
      }
    }
    testID="checkbox"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#919191",
              "fontSize": 24,
            },
            Object {
              "minWidth": 24,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#1F2532",
            "fontFamily": "IBMPlexSans-SemiBold",
            "fontSize": 26.25,
            "fontWeight": "bold",
            "lineHeight": 35.625,
            "margin": 0,
            "marginLeft": 10,
            "marginRight": 10,
            "padding": 0,
          }
        }
        testID="checkboxTitle"
      >
        Option D
      </Text>
    </View>
  </View>,
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 3,
        "borderWidth": 1,
        "margin": 0,
        "marginLeft": 0,
        "marginRight": 10,
        "opacity": 1,
        "padding": 10,
        "paddingLeft": 0,
        "paddingRight": 0,
      }
    }
    testID="checkbox"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#919191",
              "fontSize": 24,
            },
            Object {
              "minWidth": 24,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#1F2532",
            "fontFamily": "IBMPlexSans-SemiBold",
            "fontSize": 26.25,
            "fontWeight": "bold",
            "lineHeight": 35.625,
            "margin": 0,
            "marginLeft": 10,
            "marginRight": 10,
            "padding": 0,
          }
        }
        testID="checkboxTitle"
      >
        Option E
      </Text>
    </View>
  </View>,
]
`;

exports[`choicesInput (DropDown) should render a list of choices as dropdown 1`] = `
Array [
  <Text
    accessibilityHint="Single Choice - only one answer option can be selected."
    style={
      Object {
        "color": "#1F2532",
        "fontFamily": "IBMPlexSans-SemiBold",
        "fontSize": 32.8125,
        "lineHeight": 42.1875,
        "marginBottom": 5,
        "marginLeft": 0,
        "marginTop": 5,
      }
    }
  />,
  <View>
    <RNCPicker
      items={
        Array [
          Object {
            "label": "Option A",
            "testID": undefined,
            "textColor": undefined,
            "value": "Option A",
          },
          Object {
            "label": "Option B",
            "testID": undefined,
            "textColor": undefined,
            "value": "Option B",
          },
          Object {
            "label": "Option C",
            "testID": undefined,
            "textColor": undefined,
            "value": "Option C",
          },
          Object {
            "label": "Option D",
            "testID": undefined,
            "textColor": undefined,
            "value": "Option D",
          },
          Object {
            "label": "Option E",
            "testID": undefined,
            "textColor": undefined,
            "value": "Option E",
          },
        ]
      }
      numberOfLines={1}
      onChange={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
      testID="Picker"
    />
  </View>,
]
`;

exports[`choicesInput (RadioButtons) should render a list of choices as radio buttons 1`] = `
Array [
  <Text
    accessibilityHint="Single Choice - only one answer option can be selected."
    style={
      Object {
        "color": "#1F2532",
        "fontFamily": "IBMPlexSans-SemiBold",
        "fontSize": 32.8125,
        "lineHeight": 42.1875,
        "marginBottom": 5,
        "marginLeft": 0,
        "marginTop": 5,
      }
    }
  />,
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 3,
        "borderWidth": 1,
        "margin": 0,
        "marginLeft": 0,
        "marginRight": 10,
        "opacity": 1,
        "padding": 10,
        "paddingLeft": 0,
        "paddingRight": 0,
      }
    }
    testID="checkbox"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#919191",
              "fontSize": 24,
            },
            Object {
              "minWidth": 24,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#1F2532",
            "fontFamily": "IBMPlexSans-SemiBold",
            "fontSize": 26.25,
            "fontWeight": "bold",
            "lineHeight": 35.625,
            "margin": 0,
            "marginLeft": 10,
            "marginRight": 10,
            "padding": 0,
          }
        }
        testID="checkboxTitle"
      >
        Option A
      </Text>
    </View>
  </View>,
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 3,
        "borderWidth": 1,
        "margin": 0,
        "marginLeft": 0,
        "marginRight": 10,
        "opacity": 1,
        "padding": 10,
        "paddingLeft": 0,
        "paddingRight": 0,
      }
    }
    testID="checkbox"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#919191",
              "fontSize": 24,
            },
            Object {
              "minWidth": 24,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#1F2532",
            "fontFamily": "IBMPlexSans-SemiBold",
            "fontSize": 26.25,
            "fontWeight": "bold",
            "lineHeight": 35.625,
            "margin": 0,
            "marginLeft": 10,
            "marginRight": 10,
            "padding": 0,
          }
        }
        testID="checkboxTitle"
      >
        Option B
      </Text>
    </View>
  </View>,
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 3,
        "borderWidth": 1,
        "margin": 0,
        "marginLeft": 0,
        "marginRight": 10,
        "opacity": 1,
        "padding": 10,
        "paddingLeft": 0,
        "paddingRight": 0,
      }
    }
    testID="checkbox"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#919191",
              "fontSize": 24,
            },
            Object {
              "minWidth": 24,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#1F2532",
            "fontFamily": "IBMPlexSans-SemiBold",
            "fontSize": 26.25,
            "fontWeight": "bold",
            "lineHeight": 35.625,
            "margin": 0,
            "marginLeft": 10,
            "marginRight": 10,
            "padding": 0,
          }
        }
        testID="checkboxTitle"
      >
        Option C
      </Text>
    </View>
  </View>,
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 3,
        "borderWidth": 1,
        "margin": 0,
        "marginLeft": 0,
        "marginRight": 10,
        "opacity": 1,
        "padding": 10,
        "paddingLeft": 0,
        "paddingRight": 0,
      }
    }
    testID="checkbox"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#919191",
              "fontSize": 24,
            },
            Object {
              "minWidth": 24,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#1F2532",
            "fontFamily": "IBMPlexSans-SemiBold",
            "fontSize": 26.25,
            "fontWeight": "bold",
            "lineHeight": 35.625,
            "margin": 0,
            "marginLeft": 10,
            "marginRight": 10,
            "padding": 0,
          }
        }
        testID="checkboxTitle"
      >
        Option D
      </Text>
    </View>
  </View>,
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 3,
        "borderWidth": 1,
        "margin": 0,
        "marginLeft": 0,
        "marginRight": 10,
        "opacity": 1,
        "padding": 10,
        "paddingLeft": 0,
        "paddingRight": 0,
      }
    }
    testID="checkbox"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#919191",
              "fontSize": 24,
            },
            Object {
              "minWidth": 24,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#1F2532",
            "fontFamily": "IBMPlexSans-SemiBold",
            "fontSize": 26.25,
            "fontWeight": "bold",
            "lineHeight": 35.625,
            "margin": 0,
            "marginLeft": 10,
            "marginRight": 10,
            "padding": 0,
          }
        }
        testID="checkboxTitle"
      >
        Option E
      </Text>
    </View>
  </View>,
]
`;
